AnnoDesigner/AnnoCanvas.xaml.cs

```diff
@@ -1693,6 +1814,7 @@ protected override void OnKeyDown(KeyEventArgs e)
             switch (e.Key)
             {
                 case Key.Delete:
+                    // TODO: This currently doesn't work, will need to fix up.
                     // remove all currently selected objects from the grid and clear selection
                     actionManager.PerformAction(new RemovePlacedObjects(CloneList(SelectedObjects)));
```

Comment:
Remove method of the list uses HashCode and then Equals to find the correct object to remove. But here you are deep cloning the list, so the items will not be equal unless you also override those. But I'd suggest just not cloning the items of the list.

Reply:
Ah that makes sense. I need a way to identify objects uniquely (will be more important for any action which changes an objects properties like position/colour/rotation). Sounds like this is a tougher problem than I originally thought.